src/core/reactor_backend.cc

```diff
@@ -331,32 +330,41 @@ aio_general_context::~aio_general_context() {
 }
 
 void aio_general_context::queue(linux_abi::iocb* iocb) {
-    assert(last < end);
-    *last++ = iocb;
+    if (iocbs.size() < iocbs.capacity()) {
+        iocbs.push_back(iocb);
+    } else {
+        iocbs_backlog.push_back(iocb);
+    }
 }
 
 size_t aio_general_context::flush() {
-    auto begin = iocbs.get();
-    using clock = std::chrono::steady_clock;
-    constexpr clock::time_point no_time_point = clock::time_point(clock::duration(0));
-    auto retry_until = no_time_point;
+    auto begin = iocbs.data();
+    auto last = iocbs.data() + iocbs.size();
     while (begin != last) {
         auto r = io_submit(io_context, last - begin, begin);
-        if (__builtin_expect(r > 0, true)) {
-            begin += r;
-            continue;
-        }
-        // errno == EAGAIN is expected here. We don't explicitly assert that
-        // since the assert below prevents an endless loop for any reason.
-        if (retry_until == no_time_point) {
-            // allow retrying for 1 second
-            retry_until = clock::now() + 1s;
-        } else {
-            assert(clock::now() < retry_until);
+        if (__builtin_expect(r <= 0, false)) {
```

Comment:
I am happy to change it here but given this is Benny's commits it's probably best to comment the same on the original PR. Otherwise it will just get lost sooner or later.

Reply:
I see, good point about possible conflicts. I'll comment there.